# react-dom/no-namespace

## Rule category

Correctness.

## What it does

Enforces the absence of a `namespace` in React elements.

## Why is this bad?

Namespaces, such as with `svg:circle` are not supported in React.

## Examples

### Failing

```tsx {1,2}
<ns:TestComponent />;
<Ns:TestComponent />;
```

### Passing

```tsx
<TestComponent />;
<testComponent />;
```
