# no-leaked-conditional-rendering

ðŸ’­ This rule requires type information.

## Rule category

Suspicious.

## What it does

Prevents problematic leaked values from being rendered.

## Why is this bad?

Using the && operator to render some element conditionally in JSX can cause unexpected values being rendered, or even crashing the rendering.

## Examples

In React, you might end up rendering unexpected values like 0 or NaN. In React Native, your render method will even crash if you render these values:

```tsx {5,9,13}
function MyComponent() {
  return (
    <>
      {/* Warning: Possible unexpected value will be rendered */}
      {0 && <Something />}
      {/* React: renders undesired 0 */}
      {/* React Native: crashes ðŸ’¥ */}

      {/* Warning: Possible unexpected value will be rendered */}
      {NaN && <Something />}
      {/* React: renders undesired NaN */}
      {/* React Native: crashes ðŸ’¥ */}

      {/* Warning: Possible unexpected value will be rendered */}
      {"" && <Something />}
      {/* React: renders nothing */}
      {/* React Native, with React < 18: crashes ðŸ’¥ */}
    </>
  );
}
```

This can be avoided by:

- coercing the conditional to a boolean: `{!!someValue && <Something />}`
- transforming the binary expression into a ternary expression which returns null for falsy values: `{someValue ? <Something /> : null}`

### Failing

```tsx {2}
function MyComponent({ count, title }) {
  return <div>{count && title}</div>;
}
```

```tsx {2}
function MyComponent({ count }) {
  return <div>{count && <span>There are {count} results</span>}</div>;
}
```

```tsx {2}
function MyComponent({ elements }) {
  return <div>{elements.length && <List elements={elements} />}</div>;
}
```

```tsx {4}
function MyComponent({ nestedCollection }) {
  return (
    <div>
      {nestedCollection.elements.length && <List elements={nestedCollection.elements} />}
    </div>
  );
}
```

```tsx {2}
function MyComponent({ elements }) {
  return <div>{elements[0] && <List elements={elements} />}</div>;
}
```

```tsx {2}
function MyComponent({ numberA, numberB }) {
  return <div>{(numberA || numberB) && <Results>{numberA + numberB}</Results>}</div>;
}
```

```tsx {4}
// If the condition is a boolean value, this rule will report the logical expression
// since it can't infer the type of the condition.
function MyComponent({ someBool }) {
  return <div>{someBool && <Results>{numberA + numberB}</Results>}</div>;
}
```

### Passing

```tsx
function MyComponent({ elements }) {
  return <div>{elements}</div>;
}
```

```tsx
// An OR condition it's considered valid since it's assumed as a way
// to render some fallback if the first value is falsy, not to render something conditionally.
function MyComponent({ customTitle }) {
  return <div>{customTitle || defaultTitle}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>There are {elements.length} elements</div>;
}
```

```tsx
function MyComponent({ elements, count }) {
  return <div>{!count && "No results found"}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>{!!elements.length && <List elements={elements} />}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>{Boolean(elements.length) && <List elements={elements} />}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>{elements.length > 0 && <List elements={elements} />}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>{elements.length ? <List elements={elements} /> : null}</div>;
}
```

```tsx
function MyComponent({ elements }) {
  return <div>{elements.length ? <List elements={elements} /> : <EmptyList />}</div>;
}
```

## Further Reading

- [react.dev: Conditional Rendering](https://react.dev/learn/conditional-rendering)
