# react-ts/jsx/prefer-shorthand-boolean

## Rule Details

This rule enforces the use of shorthand syntax for boolean attributes.

Examples of **incorrect** code for this rule:

```tsx
<Component hidden={true} disabled={false} />;
```

Examples of **correct** code for this rule:

```tsx
<Component hidden disabled />;
```

## Rule Options

This rule has no options.
